model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py

```diff
@@ -67,35 +63,29 @@ def _build_input_tensors_list(node: BaseNode,
             _input_tensors = node_to_output_tensors_dict[ie.source_node]
             input_tensors.append(_input_tensors)
         input_tensors = [tensor for tensor_list in input_tensors for tensor in tensor_list]  # flat list of lists
-        input_tensors = node.insert_positional_weights_to_input_list(input_tensors)
-        # convert inputs from positional weights (numpy arrays) to tensors. Must handle each element in the
-        # list separately, because in FX the tensors are FX objects and fail to_torch_tensor
-        input_tensors = [to_torch_tensor(t, numpy_type=t.dtype) if isinstance(t, np.ndarray) else t
-                         for t in input_tensors]
     return input_tensors
 
 
 def _merge_inputs(_node: BaseNode, input_tensors: List, op_call_args: List,
-                  is_op_quantize_wrapper: bool) -> List:
+                  tensor_input_indices: List = None) -> List:
     """
-    Merge input tensors list with op_call_args, according to correct order.
+    Merge input tensors list with positional weights and op_call_args, according to correct order.
 
     Args:
         _node: The node the inputs are for.
         input_tensors: activation input tensors to node.
         op_call_args: framework node call args.
-        is_op_quantize_wrapper: Whether the func_op is a PytorchQuantizationWrapper or not.
+
     Returns:
         Combined list of input_tensors and op_call_args.
     """
     if isinstance(_node, FunctionalNode) and _node.tensor_input_indices:
         _input_list = op_call_args.copy()
-        if is_op_quantize_wrapper:
-            _input_list = input_tensors + _input_list
-        else:
-            assert len(_node.tensor_input_indices) == len(input_tensors), 'Mismatch between input tensors and indices'
-            for i, t in zip(_node.tensor_input_indices, input_tensors):
-                _input_list.insert(i, t)
+        if tensor_input_indices is None:
+            tensor_input_indices = _node.tensor_input_indices
+        assert len(tensor_input_indices) == len(input_tensors), 'Mismatch between input tensors and indices'
```

Comment:
I don't know if a logger is needed, but please update the message with the mismatched values.